software/cnvkit/main.nf

```diff
@@ -0,0 +1,65 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+def VERSION = '0.9.8'
+
+process CNVKIT {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+	
+    conda (params.enable_conda ? "bioconda::cnvkit=0.9.8=0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/conda:0.9.8--py_0"
+    } else {
+        container "quay.io/biocontainers/cnvkit:0.9.8--py_0"
+    }
+
+
+    input:
+    tuple val(meta), path(tumourbam), path(normalbam)
+    path fasta
+    path annotationfile
+
+
+    output:
+    tuple val(meta), path("*.cnn"), emit: cnn
+    tuple val(meta), path("*.targetcoverage.cnn"), emit: cnn
+    tuple val(meta), path("*.antitargetcoverage.cnn"), emit: cnn
+    tuple val(meta), path("*.bed"), emit: bed
+    tuple val(meta), path("*.target.bed"), emit: bed
+    tuple val(meta), path("*.antitarget.bed"), emit: bed
+    tuple val(meta), path("*.cnr"), emit: cnr
+    tuple val(meta), path("*.cns"), emit: cns
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}.${options.suffix}" : "${meta.id}"
+    if (meta.test_with_normal) {
+        """
+        cnvkit.py batch $options.args $tumourbam 
+          --normal $normalbam \\
+          --method wgs \\
+          --fasta $reffasta \\

```

Comment:
```suggestion
          --fasta $fasta \\
```

Reply:
change done!

Reply:
> The `version` command can be used to get the version. cf suggestions.
> I think the `--method wgs` and `--output-reference reference.cnn --output-dir output` should go out of the process, and go in `options.args`.
> Only thing I can see missing would be a `--targets ${target_bed}` option.
> I'm not sure about the `meta.test_with_normal`
> Are you already using something similar in a pipeline?
> I would rather separate the 2 usages in 2 processes that would take different input.
> But actually not sure what the consensus is on that matter, so I'll ping @drpatelh for some wisdom on that.




Hi Maxime,
> I have deleted `--method wgs` and `--output-reference reference.cnn --output-dir output`  from the process. Where should it go then? in the test.yml?
> --targets is missing because currently this module file is for WGS data which does not require target.bed. It is a target panel data that requires --targets. Once the module file is successful for the WGS data, I will add --target for the target panel also in a different process